account_invoice_update_stock_valuation/tests/test_account_invoice_update_stock_valuation.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields
+from odoo.tests import Form, SavepointCase
+
+
+class TestAccountInvoiceUpdateStockValuation(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.partner = cls.env.ref("base.res_partner_1")
+        cls.product_6 = cls.env.ref("product.product_product_6")
+        cls.product_7 = cls.env.ref("product.product_product_7")
+
+    def test_update_stock_valuation_from_invoice(self):
+        products = self.product_6
+        purchase = self._create_purchase_order(self.partner, products)
+        invoice = self._create_invoice_from_purchase(purchase)
+        invoice_line = invoice.invoice_line_ids[0].with_context(
+            check_move_validity=False
+        )
+        invoice_line.price_unit += 10.0
+        invoice._recompute_dynamic_lines()
+        self.assertEqual(invoice_line.stock_valuation_ok, False)
+        invoice.invoice_date = fields.Date.today()
+        invoice.action_post()
+        invoice_line._compute_stock_valuation_ok()
+        self.assertEqual(invoice_line.stock_valuation_ok, True)
+        svl = self.env["stock.valuation.layer"].search(
+            [
+                ("product_id", "=", invoice_line.product_id.id),
+                ("account_move_id", "=", invoice.id),
+                ("company_id", "=", invoice_line.company_id.id),
+                ("value", "=", invoice_line.price_unit),
+            ]
+        )
+        self.assertTrue(svl)
+
```

Comment:
Can you add tests where the invoice uom qty is different ?

def test_update_stock_valuation_from_invoice_case_2(self):
     purchase order with 12 unit
     invoice with  1 Dozens

def test_update_stock_valuation_from_invoice_case_2(self):
     purchase order with 1 Dozens
     invoice with  1 Dozens

Reply:
@sebastienbeau , it's done.